### PR TITLE
[COMCTL32][USER32] Button: Fix DLGC_... handling

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Based on KRosUser's `button.patch`.
JIRA issue: [CORE-17210](https://jira.reactos.org/browse/CORE-17210)

## Proposed changes

- Fix `DLGC_...` handling by using `&` operator in `BUTTON_CheckAutoRadioButton` in `button.c`.
- Fix `DLGC_...` handling by using `&` operator in `IsDialogMessageW` in `dialog.c`.
- `BM_CLICK`'s `wParam` must be zero.

## TODO

- [x] Do tests.

### CODE REVIEW

File: win32ss/user/user32/controls/button.c [hunk 1]
```diff
@@ -1196,7 +1196,11 @@ static void BUTTON_CheckAutoRadioButton( HWND hwnd )
     do
     {
         if (!sibling) break;
+#ifdef __REACTOS__
+        if ((SendMessageW( sibling, WM_GETDLGCODE, 0, 0 ) & (DLGC_BUTTON | DLGC_RADIOBUTTON)) == (DLGC_BUTTON | DLGC_RADIOBUTTON))
+#else
         if (SendMessageW( sibling, WM_GETDLGCODE, 0, 0 ) == (DLGC_BUTTON | DLGC_RADIOBUTTON))
```
Comment (SergeGautherie) [inline win32ss/user/user32/controls/button.c:1202]: Is
https://github.com/reactos/reactos/blob/f0179741d1a5aa67ee62a0f33b07962b63a5fad1/win32ss/user/user32/windows/dialog.c#L2631
affected too?

Comment (katahiromz) [reply win32ss/user/user32/controls/button.c:1202]: Thanks.

File: win32ss/user/user32/windows/dialog.c [hunk 2]
```diff
@@ -2635,12 +2635,14 @@ IsDialogMessageW(
...
-                 if (hwndNext && SendMessageW( hwndNext, WM_GETDLGCODE, lpMsg->wParam, (LPARAM)lpMsg ) == (DLGC_BUTTON | DLGC_RADIOBUTTON))
+                 if (hwndNext &&
+                     ((SendMessageW(hwndNext, WM_GETDLGCODE, lpMsg->wParam, (LPARAM)lpMsg) &
+                       (DLGC_BUTTON | DLGC_RADIOBUTTON)) == (DLGC_BUTTON | DLGC_RADIOBUTTON)))
                  {
                      SetFocus( hwndNext );
                      if ((GetWindowLongW( hwndNext, GWL_STYLE ) & BS_TYPEMASK) == BS_AUTORADIOBUTTON &&
                          SendMessageW( hwndNext, BM_GETCHECK, 0, 0 ) != BST_CHECKED)
-                         SendMessageW( hwndNext, BM_CLICK, 1, 0 );
+                         SendMessageW(hwndNext, BM_CLICK, 0, 0);
```
Comment (SergeGautherie) [inline win32ss/user/user32/windows/dialog.c:2645]: Would the following work and be more consistent?
```suggestion
                         SendMessageW(hwndNext, BM_SETCHECK, BST_CHECKED, 0);
```

Comment (katahiromz) [reply win32ss/user/user32/windows/dialog.c:2645]: Can `BN_CLICKED` be generated by that?

Comment (SergeGautherie) [reply win32ss/user/user32/windows/dialog.c:2645]: > Can `BN_CLICKED` be generated by that?

I would guess it cannot. My question was more like 'is that needed'? Anyway.

--

> `BM_CLICK`'s `wParam` must be zero.

@katahiromz, Are the followings affected too?
https://github.com/reactos/reactos/blob/d55add359cc96b38ce4e9ebf4eee845dfd64ba69/base/applications/charmap/settings.c#L96
https://github.com/reactos/reactos/blob/d55add359cc96b38ce4e9ebf4eee845dfd64ba69/modules/rostests/winetests/user32/msg.c#L6834

Comment (katahiromz) [reply win32ss/user/user32/windows/dialog.c:2645]:
```c
SendDlgItemMessage(hCharmapDlg, IDC_CHECK_ADVANCED, BM_CLICK, MF_CHECKED, 0);
```
Fixed.

```c
{ BM_CLICK, sent|wparam|lparam, 1, 0 }, 
```
I'm not sure.

**Pull request merged**